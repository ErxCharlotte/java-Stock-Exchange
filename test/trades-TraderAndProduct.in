ADD trader1 100.00
ADD trader2 20.00
ADD trader3 300
IMPORT trader1 ABC 100
IMPORT trader1 CDE 1000
SELL trader1 ABC 100 5
BUY trader2 CDE 100 10
BUY trader2 ABC 10 15
SELL trader1 CDE 500 5
BUY trader2 CDE 400 5
TRADES TRADER trader4
TRADES TRADER trader3
TRADES TRADER trader1
TRADES PRODUCT LOL
TRADES PRODUCT CDE
EXIT