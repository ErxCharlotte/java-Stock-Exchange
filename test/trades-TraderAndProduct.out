$ Success.
$ Success.
$ Success.
$ Trader now has 100.00 units of ABC.
$ Trader now has 1000.00 units of CDE.
$ No trades could be made, order added to sell book.
$ No trades could be made, order added to buy book.
$ Product bought in entirety, trades as follows:
trader1->trader2: 10.00xABC for $5.00.
$ Product sold in part, trades as follows:
trader1->trader2: 100.00xCDE for $10.00.
$ Product bought in entirety, trades as follows:
trader1->trader2: 400.00xCDE for $5.00.
$ No such trader in the market.
$ No trades have been completed by trader.
$ trader1->trader2: 10.00xABC for $5.00.
trader1->trader2: 100.00xCDE for $10.00.
trader1->trader2: 400.00xCDE for $5.00.
$ No trades have been completed with given product.
$ trader1->trader2: 100.00xCDE for $10.00.
trader1->trader2: 400.00xCDE for $5.00.
$ Have a nice day.